src/selectors/entities/users.test.js

```diff
@@ -443,35 +443,106 @@ describe('Selectors.Users', () => {
             [testUser2.id]: testUser2,
         };
 
-        const newTestState = {
+        let newTestState = {
             entities: {
                 users: {profiles: newProfiles},
-                preferences: {myPreferences: {}},
-                general: {config: {TeammateNameDisplay: General.TEAMMATE_NAME_DISPLAY.SHOW_NICKNAME_FULLNAME}},
+                general: {
+                    config: {
+                        TeammateNameDisplay: General.TEAMMATE_NAME_DISPLAY.SHOW_NICKNAME_FULLNAME, 
+                        LockTeammateNameDisplay: 'false', 
+                    }
+                },
+                preferences: {
+                    myPreferences: {
+                        [`${Preferences.CATEGORY_DISPLAY_SETTINGS}--${Preferences.NAME_NAME_FORMAT}`]: {
+                            value: General.TEAMMATE_NAME_DISPLAY.SHOW_NICKNAME_FULLNAME,
+                        },
+                    },
+                }
             },
         };
 
         // should match full name, since nickname is not present
-        assert.deepEqual(Selectors.makeGetDisplayName()(newTestState, testUser1.id), 'First Last');
+        assert.deepStrictEqual(Selectors.makeGetDisplayName()(newTestState, testUser1.id), 'First Last');
 
         // should match nickname since it's present
-        assert.deepEqual(Selectors.makeGetDisplayName()(newTestState, testUser2.id), 'nick2');
+        assert.deepStrictEqual(Selectors.makeGetDisplayName()(newTestState, testUser2.id), 'nick2');
 
-        newTestState.entities.general = {config: {TeammateNameDisplay: General.TEAMMATE_NAME_DISPLAY.SHOW_USERNAME}};
+        newTestState = {
```

Comment:
Should we add a subtest to group and avoid having to redefine `newTestState`? This would also help below with the other `*config*` changes and subsequent testing.